src/store/modules/app-gateway-svc/states.ts

```diff
@@ -0,0 +1,7 @@
+import { ModuleState } from './types';
+
+// eslint-disable-next-line import/prefer-default-export
+export const state: ModuleState = {
+  grpcClient: undefined as any,
+  httpHeader: undefined as any,
```

Comment:
We can get rid of `types.ts`.

Why not just include the interface here to reduce the file to manage. The boiler plating is getting confusing.

```ts
import { AppGatewayServiceClient, grpc } from '../../root-types';

export interface ModuleState {
  grpcClient?: AppGatewayServiceClient;
  httpHeader?: grpc.Metadata;
}

export const state: ModuleState = {};
```

Reply:
> why `undefined as any`? We should make them as optional properties with `?`

Because I have to initialize it with something and type null or undefined is not a valid type for the type I defined it as. So `undefined as any` works for any type.

Reply:
> I do not understand how `states.ts` and `types.ts`. Why do we need both?

That's because this module is very small, so it might be hard to see the difference. This module's only purpose is to initialize client and http header when app.vue starts (atm).

`state.ts` only contain state of the module, `types.ts` includes any interfaces and other typings that the module will need.

Reply:
> We can get rid of `types.ts`.
> 
> Why not just include the interface here to reduce the file to manage. The boiler plating is getting confusing.
> 
> ```ts
> import { AppGatewayServiceClient, grpc } from '../../root-types';
> 
> export interface ModuleState {
>   grpcClient?: AppGatewayServiceClient;
>   httpHeader?: grpc.Metadata;
> }
> 
> export const state: ModuleState = {};
> ```

I was thinking of that too. The file could potentially get large, but I agree its confusing.

Reply:
If it is confusing then we should get rid of it?